com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/managers/BazelBuildSupport.java

```diff
@@ -0,0 +1,156 @@
+    private static final String BAZEL_PROJECT_FILE_SUFIX = ".bazelproject";
+    private static final List<String> WATCH_FILE_PATTERNS =
+            Arrays.asList("**/BUILD", "**/WORKSPACE", "**/*.bazelproject");
+
+    @Override
+    public boolean applies(IProject project) {
+        try {
+            return project != null && project.hasNature(BazelNature.BAZEL_NATURE_ID);
+
+        } catch (CoreException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public void update(IProject project, boolean force, IProgressMonitor monitor) throws CoreException {
+        try {
+            updateInternal(project, force, monitor);
+
+        } catch (CoreException ex) {
+            throw ex;
+
+        } catch (AssertionFailedException ex) {
+            // Bazel can't work with the provided project.
+            // Just skip it.
+
+        } catch (NoSuchElementException ex) {
+            // No bazel importers found. This is definitely illegal situation
+            // which is not clear how to solve right now.
+            // Just skip it.
+        }
+
+    }
+
+    protected void updateInternal(IProject project, boolean force, IProgressMonitor monitor) throws CoreException {
+
+        Assert.isTrue(applies(project));
+
+        final IProjectImporter importer = obtainBazelImporter().get();
+
+        //TODO is false becasue rootFolder is NULL
+        Assert.isTrue(importer.applies(monitor));
+
+        importer.importToWorkspace(monitor);
+    }
+
+    @Override
+    public boolean isBuildFile(IResource resource) {
+        return resource != null && resource.getProject() != null && resource.getType() == IResource.FILE
+                && (resource.getName().endsWith(BAZEL_PROJECT_FILE_SUFIX) || resource.getName().equals(BUILD_FILE)
+                        || resource.getName().equals(BUILD_FILE));
+    }
+
+    @Override
+    public boolean isBuildLikeFileName(String fileName) {
+        return fileName.endsWith(BAZEL_PROJECT_FILE_SUFIX) || fileName.equals(BUILD_FILE)
+                || fileName.equals(BUILD_FILE);
+    }
+
+    @Override
+    public List<String> getWatchPatterns() {
+        return WATCH_FILE_PATTERNS;
+    }
+
+    @Override
+    public boolean fileChanged(IResource resource, CHANGE_TYPE changeType, IProgressMonitor monitor)
+            throws CoreException {
+        if (resource == null || !applies(resource.getProject())) {
+            return false;
+        }
+        return IBuildSupport.super.fileChanged(resource, changeType, monitor) || isBuildFile(resource);
+    }
+
+    ///////
+
+    private Optional<IProjectImporter> obtainBazelImporter() {
+        return ExtensionsExtractor.<IProjectImporter>extractOrderedExtensions(IConstants.PLUGIN_ID, "importers")
+                .stream().filter(importer -> importer instanceof BazelProjectImporter).findFirst();
+    }
+
+    @Override
+    public String buildToolName() {
+        return "Bazel";
```

Comment:
Extract this as a static final variable (like `BUILD_TOOL_NAME` or somewhat) pls